### PR TITLE
Ignore comments in an ignored block

### PR DESCRIPTION
**Pull request by Janther**
if any node has the `ignore-prettier` comment, we will iterate through the whole object and mark the comments as printed.

I also took the liberty to update the dependencies and remove most listing errors in the test engine that we had.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,22 +61,22 @@
...
     "dir-to-object": "^2.0.0",
-    "emoji-regex": "^9.0.0",
+    "emoji-regex": "^9.2.1",
     "escape-string-regexp": "^4.0.0",
-    "prettier": "^2.0.5",
-    "semver": "^7.3.2",
+    "prettier": "^2.2.1",
+    "semver": "^7.3.4",
     "solidity-comments-extractor": "^0.0.4",
     "string-width": "^4.2.0"
```
Comment (mattiaerre) [inline package.json:81]: did you have to update all these deps yourself? have they been missed by dependabot? really?

File: tests_config/helpers.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+function read(filename) {
+  return fs.readFileSync(filename, 'utf8');
+}
+
+function parse(string, opts) {
+  return prettier.__debug.parse(string, opts, /* massage */ true).ast;
+}
+
+module.exports = { mergeDefaultOptions, parse, prettyprint, raw, read };
```
Comment (mattiaerre) [inline tests_config/helpers.js:44]: good idea to move them our into a separate module 👍

File: tests_config/run_spec.js [hunk 3]
```diff
@@ -73,45 +76,4 @@ function run_spec(dirname, options) {
...
-function mergeDefaultOptions(parserConfig) {
-  return Object.assign(
-    {
-      plugins: [path.dirname(__dirname)],
-      printWidth: 80
-    },
-    parserConfig
-  );
-}
+global.run_spec = runSpec;
```
Comment (mattiaerre) [inline tests_config/run_spec.js:79]: nice refactoring

File: tests_config/run_spec.js [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
-
 const fs = require('fs');
 const path = require('path');
+const {
+  read,
+  mergeDefaultOptions,
+  prettyprint,
+  raw,
+  parse
+} = require('./helpers');
```
Comment (mattiaerre) [inline tests_config/run_spec.js:10]: I'd also alpha import here but I'm also ok to do it in a next PR

**Review by mattiaerre — APPROVED** [review]
that's fantastic, thanks @Janther

File: package.json (hunk 1 above)
Comment (Janther) [reply package.json:81]: yeah dependabot does the heavy lifting for critical dependencies I believe. also I did not have any error pop up when I created the new tests and only started failing when I updated the dependancies. maybe the latest prettier did the trick.

Comment (mattiaerre) [reply package.json:81]: splendid, I'll merge and deploy thanks @Janther

**Pull request merged**